src/preprocessing/passes/unconstrained_simplifier.cpp

```diff
@@ -236,6 +236,10 @@ void UnconstrainedSimplifier::processUnconstrained()
               break;
             }
           }
+          if (parent[0].getType().getCardinality().isOne())
+          {
+            break;
+          }
```

Comment:
Do we still need the check of `dt.isRecursiveSingleton(tn.toType())` below or does this check subsume that?

Reply:
We still need it.  This is a really tricky case: this handles infinite streams having uninterpreted sorts (which may or may not have cardinality one, which cause the stream to have either infinite cardinality or cardinality one).